### ISSUE TITLE
Announcement: Changing License from MIT to GPL-3

### ISSUE THREAD

**Issue by timgrossmann** [InstaPy/InstaPy#2689]
Hey everyone,

In the recent weeks and months I received quite some requests from less and more established, actual, companies asking me to help them build GUIs, more bots or similar thing based off of InstaPy. When asking more closely, I, a lot of times, got the answer that they actually use InstaPy as the core tool and "just build some features and a GUI" on top of it.

There are a lot of great people around here who work really hard to keep this project going and make sure that everything works and people get new features. 
**They do this in their free time, not getting a single cent for it.**
If you use this tool, please don't forget this.

The idea of making this project "MIT"-licensed was that I believed that if people/companies used it commercially, they would contribute the changes back to the tool in order for the community to also profit from them using it.
I, sadly, had to realise that it does not work this way. A lot of the smaller "companies" and people who make a little bit of money with it actually do contribute and grow this project.
**However** the bigger "players" are pretty selfish, they keep their improvements and changes to themselves, absolutely ignoring the fact what made and makes this project what it is. 
What bugs me most is that those are the "users" of InstaPy which would have the money and resources to put some hours into improving InstaPy.

Therefore I will change the license of this project, starting on Monday 16th of August 2018, to GPL-3.
The most important change for me is that you will have to open source the changes you makes to your fork of InstaPy again.
Pretty much everything else will stay similar, you will still be able to legally use this tool 

> If you want to know what "restrictions" this new license has, you can read it here: https://tldrlegal.com/license/gnu-general-public-license-v3-(gpl-3)#summary

> Oh, btw. those license changes will only affect InstaPy versions greater than the actual license change commit. Just to point this out here.

What do you guys think about this?
Would love to get some opinions about this.

> @converge @uluQulu @sionking @CharlesCCC

**Comment by uluQulu** [comment]
Hello @timgrossmann
How sad it is.. they should really share all of the foundings with the community!

MIT or any other GNU license has no any difference at my point ⚖
If I was to need a license, I would either get a custom proprietary license or not license at all.

And it seems, you have reasons to decide it.
That's why me applauses your decision and wishes a best product 👏🏼👏🏼👏🏼

----------

Cheers 🤩
--------

**Comment by lightseb** [comment]
Clap clap !! Great Decision !!! I wish you the best too ;)

"If I was to need a license, I would either get a custom proprietary license or not license at all." Same here !

**Comment by imjustin** [comment]
I think changing the license is a great idea and i'm all for it. with that being said, if the hope is that companies building on top of instapy will publish out their changes/features as well, i don't think that'll realistically happen. plus from what you shared it doesn't seem like they want to build the features themselves; they want you to build it?

love the community and what you've built, wish you all the best :)

**Comment by Malusovium** [comment]
This is great!

**Comment by JosteinFraNorge** [comment]
This is great.

On a side note I think that you should go public with the names of the bigger companies as well :-)

**Comment by sionking** [comment]
I don't really understand in those things but I can tell for sure. If you make money from instapy you need either to contribute time or money.

**Comment by Areahints** [comment]
i love open-source so i know what it feels like to see a company take community work and spin it off as some sort of fantastic new product! let all users  contribute either time or money!

**Comment by timgrossmann** [comment]
Thank you very much for the reassurance on this topic! 👍 
This is really important to me, so I'm glad your opinions are the same than mine.

@imjustin More like 
*"love the community and what ~you've~ we've built, wish you all the best :)"* 😉 

@herrgutt I thought about this too, however I decided not to disclose any names due to the fact that I want to give them a chance to "do their part" in making InstaPy better without "attacking" them. Maybe some of those bigger players realise that it's not the right way to go to simply use something a whole community built and just make money out of it without either donating some of it to be used in the project or help out by contributing 🙏 

Considering all your comments, I really have to think hard about what license I'll take. 
I think GPL is interesting since they are "forced" to open source their changes again. 
However they are still allowed to make money out of it. If we completely forbid commercial use, we might never get some contributions/donations from bigger users since they simply won't use it.
Tough decision and I can absolutely understand your preference for a proprietary or none license at all @uluQulu @lightseb

**Comment by imjustin** [comment]
There may be a model where companies/clients who are interested in using instapy for commercial use can sponsor/fund the creation of features, similar to a bounty system?

**Comment by sionking** [comment]
But this kind of application is not the most "legal", why do you think those big companies care about the licence? as long as the code is shown to public you can take it and no one will know ... they can handle with other code  managements tools

**Comment by Malusovium** [comment]
@timgrossmann I take it that requirements/guide-lines regarding changes to the license will be added to the documentation?
Would be quite handy for someone in my position :stuck_out_tongue:

**Comment by timgrossmann** [comment]
@sionking I absolutely understand what you're saying! Even though that most likely will be the case, I think this is the right step to take in order to politely ask those parties to "contribute" something to InstaPy. 
If the license states so, maybe more people will be noticed that it would be incredible if they could contribute something to the project , back. Especially since there some companies, that seem to be quite fair and maybe just need a little "push" in the right direction 😉 

@Malusovium I'm really sorry but I don't completely understand what you're saying. Do you simply want a guide-lines file that states the changes of the license? Could you elaborate on that?

**Comment by Malusovium** [comment]
@timgrossmann My question seems rather vague indeed.
Coincidentally, licenses can be a quite vague too.

Well about the implication of the license:

What should a party provide when making modifications to the code?
Will git-commit messages suffice or should one be specifying every minute detail of changes in a 5000 lines long text.

If a party were to make a derivative project(things like instapy-tools) using InstaPy, what are the implications then?
Will a link to the source suffice? Must it then be in clear view? What would be in clear view?

TL;DR Will there be a guide included to properly comply with the license?

**Comment by timgrossmann** [comment]
@Malusovium As far as I know, the main thing with GPLv3 is that you're supposed to open source the changes you make to your copy of InstaPy. You don't have to merge them back to comply with the license. However it would be awesome if you would do that, just because you love InstaPy, of course 😉 

So there won't be any special file that asks you for things, just a simple "security" since a lot of, non-shady, companies don't want to use "copy-left" licenses like GPL.
So in the worst case, if I find out there is a bigger one using a very modified version of InstaPy, I could "force" them to open source their changes to InstaPy as well, giving me the chance to merge them back into the original project which get's maintained by the community.
